### ISSUE TITLE
Add support for iOS 15 / Xcode 13

### ISSUE THREAD

**Issue by LordParsley** [steamclock/bluejay#259]
Summary:

When using Xcode 13 to build an iOS 15 app with the Bluejay dependency, there are compiler errors:

```
Value of optional type 'CBService?' must be unwrapped to refer to member 'uuid' of wrapped base type 'CBService'
```

Steps to Reproduce:

1. Include `pod 'Bluejay'` as a Cocoapods dependency.
2. Open workspace in Xcode 13.0.
3. Try build or run the project.

Actual Result:

<img width="1508" alt="Screenshot 2021-09-30 at 13 54 15" src="https://user-images.githubusercontent.com/7753956/135451032-6cb9c60a-f252-4b3e-8616-509fce2a7928.png">

Expected Result: 

Bluejay supports the newer CoreBluetooth API which has an optional `CBService` property on `CBCharacteristic` and the build succeeds.

```
    /**
     * @property service
     *
     *  @discussion
     *      A back-pointer to the service this characteristic belongs to.
     *
     */
    weak open var service: CBService? { get }
```

Device, Build, OS:

Any simulator or device building against iOS 15.

**Comment by LordParsley** [comment]
Thanks for the great library! Any attention on this would be much appreciated! 😊 

FYI there is a PR submitted which checks the compiler version & then force-unwraps the service: https://github.com/steamclock/bluejay/pull/258.

It might be safer to change the `CharacteristicIdentifier` `service` property to an optional here https://github.com/steamclock/bluejay/blob/master/Bluejay/Bluejay/CharacteristicIdentifier.swift#L15 but that causes a lot of downstream changes where `.service` is being accessed.

A default service value could be used [but that might be bad practice because I'm not sure whether a zero UUID is reserved or even valid]:

```
    /// Create a `CharacteristicIdentifier` using a `CBCharacterstic`.
    public init(_ cbCharacteristic: CBCharacteristic) {
        self.service = ServiceIdentifier(uuid: cbCharacteristic.service?.uuid ?? CBUUID(string: "00000000-0000-0000-0000-000000000000"))
        self.uuid = cbCharacteristic.uuid
    }
```

Any discussion welcome!

**Comment by OmarSpyras** [comment]
Hi any news on adding support ?

**Comment by shamangeorge** [comment]
What is the status on this?

**Comment by nbrooke** [comment]
Opened a PR that fix this (#260), should be merged and released in at most a few days, but if anyone wants to try it out faster, you can pull from that branch in your package manager of choice.

**Comment by nbrooke** [comment]
Fixed in v0.8.9
